alpha-commons/src/main/java/at/ac/tuwien/kr/alpha/commons/programs/rules/BasicRule.java

```diff
@@ -25,23 +25,20 @@
  * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
  * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
  */
-package at.ac.tuwien.kr.alpha.core.rules;
+package at.ac.tuwien.kr.alpha.commons.programs.rules;
 
 import java.util.List;
 
 import at.ac.tuwien.kr.alpha.api.programs.literals.Literal;
-import at.ac.tuwien.kr.alpha.api.rules.Rule;
-import at.ac.tuwien.kr.alpha.api.rules.heads.Head;
+import at.ac.tuwien.kr.alpha.api.programs.rules.heads.Head;
 
 /**
  * Represents a non-ground rule or a constraint. A {@link BasicRule} has a general {@link Head}, meaning both choice heads and disjunctive
  * heads are permissible.
- * This implementation represents a rule after being parsed from a given ASP program, but before being transformed into a
- * {@link NormalRuleImpl}.
```

Comment:
Maybe I a missing something, but with this usage-comment removed, it feels like `BasicRule` has no use other than being an alternative name for the type `AbstractRule<Head>`. Why do we want to keep this type?

Reply:
Whenever I tried to streamline this inheritance hierarchy I ran into sublte issues with the involved generics. I'd like to keep it as-is for simplicities' sake.